### DISCUSSION TITLE
Deployed firebase s.p.a website shows 404 before index upon refresh

### DISCUSSION THREAD

**Discussion by ThibautJoukes** [gatsbyjs/gatsby#34442] [Help] [answered]
After I deploy my (single page application) website to firebase and refresh the page, it will load the 404 page before the index one. You can barely see it because the transition goes pretty fast, but when you spam refresh you can clearly see it. 

![ezgif-5-5ed9238e59](https://user-images.githubusercontent.com/19625435/148875291-4d6e0b0c-66de-461f-b627-5994b3158441.gif)

I tried to find anything related to this, but i couldn't. For now I can delete all the styling in 404 so it's a blank page to hide it.

I am using the default firebase configuration and added a these rewrites. But this doesn't seem to work for a s.p.a.

```
"rewrites": [
      {
        "source": "/",
        "destination": "/index.html"
      },
      {
        "source": "**",
        "destination": "/index.html"
      }
    ],
```

This only happens in production, not in development. My only option now is to use another hosting, but I still hope there is something I can do to fix this.

**Comment by ThibautJoukes** [comment, marked as answer]
When refreshing, firebase is trying to search for the “/home” directory/file and can’t find it because it's a s.p.a website. As firebase is probably only good for static sites, i'll use another hosting.

**Comment by ZoranNikolov** [comment]
Hey there, I had a similar problem. When I refresh the website on the home page (`/`), everything is fine.
But when I go to a link in the website, for example `/about`, I got a default 404 page of Firebase.

I fixed the issue configuring the firebase.json file.

Here are the settings that worked for me:

`{
	"hosting": {
		"public": "build",
		"ignore": ["firebase.json", "**/.*", "**/node_modules/**"],
		"rewrites": [
			{
				"source": "**",
				"destination": "/index.html"
			}
		]
	}
}` 

I hope it could help you, too.
